src/geoFire/geoQuery.ts

```diff
@@ -0,0 +1,526 @@
+import * as firebase from 'firebase';
+
+import { GeoCallbackRegistration } from './geoCallbackRegistration';
+import { decodeGeoFireObject, distance, encodeGeohash, geoFireGetKey, geohashQueries, validateCriteria, validateLocation } from './geoFireUtils';
+
+import { QueryCriteria } from './interfaces';
+
+/**
+ * Creates a GeoQuery instance.
+ */
+export class GeoQuery {
+  // Event callbacks
+  private _callbacks: any = { ready: [], key_entered: [], key_exited: [], key_moved: [] };
```

Comment:
Seems like this is not actually `any` typed at all. You could just let the type inference system work here (and provide hints), or provide an interface.